documentation/specs/multithreading/analyzer-spec.md

```diff
@@ -0,0 +1,219 @@
+# IMultiThreadableTask Analyzer - Design Proposal
+
+**Target MSBuild Version**: 18.1+  
+**Authors**: @JanProvaznik  
+**Last Updated**: October 1, 2025
+
+---
+
+## Executive Summary
+
+### Problem
+
+MSBuild's multithreaded execution allows tasks implementing `IMultiThreadableTask` to run concurrently. Many .NET APIs depend on process-global state (working directory, environment variables), creating race conditions:
+
+```csharp
+// Task A: Environment.CurrentDirectory = "C:\\ProjectA"
+// Task B: Environment.CurrentDirectory = "C:\\ProjectB" (races with A)
+// Task A: File.Exists("bin\\output.dll") // Resolves incorrectly!
+```
+
+### Solution
+
+Roslyn analyzer detecting unsafe API usage in `IMultiThreadableTask` with automated code fixers.
+
+### Benefits
+
+- Compile-time detection
+- Automated fixes for common patterns
+- Education through diagnostic messages
+
+---
+
+## 1. Design Overview
+
+### 1.1 Diagnostic Codes
+
+| Code | Severity | Description | Fixer |
+|------|----------|-------------|-------|
+| MSB9999 | Error | Critical APIs (no safe alternative) | No |
+| MSB9998 | Warning | APIs needing TaskEnvironment | Partial |
+| MSB9997 | Warning | File APIs needing absolute paths | Yes |
+| MSB9996 | Warning | Potentially problematic APIs | No |
```

Comment:
I think I'd like to have a distinct namespace/error code space here since these aren't exactly build-time errors.

Maybe just `MSBuildTask1234`? @baronfel?